pre-receive-hooks/block_self_merge_prs.sh

```diff
@@ -8,10 +8,9 @@
 # https://help.github.com/enterprise/admin/guides/developer-workflow/managing-pre-receive-hooks-on-the-github-enterprise-appliance/
 #
 
-if [ "$GITHUB_VIA" = "merge api" && "$GITHUB_PULL_REQUEST_AUTHOR_LOGIN" = "$GITHUB_USER_LOGIN"]; then
+if [[ "$GITHUB_VIA" = "merge api" ]] && [[ "$GITHUB_PULL_REQUEST_AUTHOR_LOGIN" = "$GITHUB_USER_LOGIN" ]]; then
```

Comment:
Thinking out loud here, if we need to match multiple `$GITHUB_VIA` values, we could use extglobs, e.g.:

```bash
shopt -s extglob
...
if "$GITHUB_VIA" = @(merge api|pull request merge api)
...
```